final_gen_solve/genmethods.cpp

```diff
@@ -22,38 +22,30 @@ void generateAdjacentBeta(PetscInt lattice_size, PetscInt i, PetscInt j, std::ve
     //q and q+1
     output[0] = 1.0;
 
-    for (PetscInt x = 1; x < N + 1; x++)
-    {
+    for (PetscInt x = 1; x < N + 1; x++){
         PetscInt twoRaisedX = 1;
-        for (int i = 0; i < x; i++)
-        {
+        for (int i = 0; i < x; i++){
```

Comment:
Is this iterating over all possible matrix elements in the Hamiltonian? That is the performance killer.